### PR TITLE
Add a Async Select that fetches options from given endpoint

### PR DESCRIPTION
**Pull request by vera-liu**
Done:
 - added a AsyncSelect component that fetches options from data endpoint, structure is similar to TableLoader
 - use AsyncSelect for users and databases (query search page and sql editor left)
 - replace DatabaseSelect with AsyncSelect
 - added specs

UI stays the same as in old Select:
screen shot 2017-01-05 at 2 50 55 pm
screen shot 2017-01-05 at 2 51 06 pm

needs-review @ascott @mistercrunch @bkyryliuk

### CODE REVIEW

File: superset/assets/spec/javascripts/sqllab/QuerySearch_spec.jsx [hunk 1]
```diff
@@ -19,13 +19,7 @@ describe('QuerySearch', () => {
...
 
   it('should have four Select', () => {
-    expect(wrapper.find(Select)).to.have.length(4);
-  });
-
-  it('updates userId on user selects change', () => {
-    wrapper.find('[name="select-user"]')
-      .simulate('change', { value: 1 });
-    expect(wrapper.state().userId).to.equal(1);
+    expect(wrapper.find(Select)).to.have.length(3);
```
Comment (ascott) [inline superset/assets/spec/javascripts/sqllab/QuerySearch_spec.jsx]: should this assert 3 or 4 selects? it was 4 before this change.

File: superset/assets/javascripts/components/AsyncSelect.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      this.setState({ options: mutator ? mutator(data) : data, isLoading: false });
+    });
+  }
+  onChange(opt) {
+    this.props.onChange(opt);
+  }
+  render() {
+    return (
+      <div>
+        <Select
```
Comment (ascott) [inline superset/assets/javascripts/components/AsyncSelect.jsx]: what do we show when the select is loading data?

Comment (vera-liu) [inline superset/assets/javascripts/components/AsyncSelect.jsx]: isLoading is true when loading the data, Select will render a spinner

File: superset/assets/spec/javascripts/sqllab/QuerySearch_spec.jsx (hunk 1 above)
Comment (ascott) [inline superset/assets/spec/javascripts/sqllab/QuerySearch_spec.jsx]: ok cool. we should update the `it` statement on L21 to read `should have three selects` in that case.

**Comment by ascott** [comment]
LGTM after updating the `it` statement in the spec.

File: superset/assets/javascripts/components/AsyncSelect.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  onChange: React.PropTypes.func.isRequired,
+  mutator: React.PropTypes.func.isRequired,
+  value: React.PropTypes.number,
+  valueRenderer: React.PropTypes.func,
+  placeholder: React.PropTypes.string,
+};
+
+const defaultProps = {
+  placeholder: 'Select ...',
+  value: null,
```
Comment (mistercrunch) [inline superset/assets/javascripts/components/AsyncSelect.jsx]: I think `null` defaults are implicit.

**Comment by mistercrunch** [comment]
Neat! This reusable component is super useful to have!

**Pull request merged**